pkg/controller/csi_handler.go

```diff
@@ -354,6 +362,22 @@ func (h *csiHandler) hasVAFinalizer(va *storage.VolumeAttachment) bool {
 	return false
 }
 
+// Checks if the PV (or) the inline-volume corresponding to the VA could have migrated from
+// in-tree to CSI.
+func (h *csiHandler) isMigratable(va *storage.VolumeAttachment) (bool, error) {
+	if va.Spec.Source.PersistentVolumeName != nil {
+		pv, err := h.pvLister.Get(*va.Spec.Source.PersistentVolumeName)
+		if err != nil {
+			return false, err
+		}
+		return h.translator.IsPVMigratable(pv), nil
+	} else if va.Spec.Source.InlineVolumeSpec != nil {
+		return va.Spec.Source.InlineVolumeSpec.CSI != nil, nil
```

Comment:
I think it should be an error if InlineVolumeSpec is set, but CSI is nil

Reply:
ack

Reply:
Done